staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go

```diff
@@ -122,8 +122,8 @@ func newAdmissionMetrics() *AdmissionMetrics {
 
 	// Admission webhook metrics. Each webhook is identified by name.
 	webhook := newMetricSet("webhook",
-		[]string{"name", "type", "operation", "rejected"},
-		"Admission webhook %s, identified by name and broken out for each operation and API resource and type (validate or admit).", false)
+		[]string{"name", "type", "operation", "rejected", "ignore_call_failure", "call_failure", "code"},
```

Comment:
I don't have a great sense for the extra cost of ~doubling the number of labels. will defer to @logicalhan / @kubernetes/sig-instrumentation-pr-reviews on these

Reply:
Doubling the number of labels isn't necessarily the issue, but how many timeseries they will cause to generate multiplicatively based on their values.

How many values are these labels likely to have? Generating a new timeseries for every HTTP code on top of these other label dimensions, for example, seems like it might result in a lot more load.

Reply:
- `ignore_call_failure`: boolean, typically only one pre-defined value (true/false) throughout a webhook configuration's life
- `call_failure`: boolean, can be either true (everything works fine) or false (network failure, webhook response is buggy/malformed)
- `code`: int, zero when the request is allowed, non-zero when the request is rejected, depends on webhook implementation-- a well-designed webhook should reject requests with reasonable & standard HTTP status code >= 400 (I imagine 400, 403, 404, 406, 415, 500, 503); a malicious webhook could reject requests with arbitrary numbers
  - perhaps we need to add some validation to bound the value

Reply:
is exact status code correlation important or could we bucket it into 4xx and 5xx?

Reply:
I could envision these response codes:
* 400, 401, 403, 404, 409, 410, 422, 429
* 500, 503

Is ~10 values for the code label problematic? If not, what if we truncated any 4xx code outside that set to 400 and any 5xx code outside that set to 500?

Reply:
@brancz wdyt?

Reply:
Exact status code correlation is not important to me. My goal is to distinguish a real apiserver internal error from 1) an error invoking the webhook and 2) webhook responding with 500. Bonus point if we can monitor more information with a low cost.

Technically my goal is achievable with one boolean label:
- `(real apiserver)internal_error`: true/false, but having such a label in a webhook metrics sounds weird to me

Other labels that are nice to have:
- `ignore_call_failure` + `call_failure`: let us know when a "rejection" is actually ignored
- `code`: I can go with two values {500, xxx} or three values {4xx, 500, 5xx}; or {4xx, 5xx} if we have the `internal_error` label

Reply:
I actually like your suggestion of the one additional label. Why not just add `apiserver_internal_error` and call it a day?

Reply:
> Is ~10 values for the code label problematic? 

Nah, it's fine.

Reply:
I realized that `apiserver_internal_error` is not sufficient to tell how many 500s in `apiserver_request_count{code="500"}` are caused by webhooks. I changed `code` to log four values: {0, 4xx, 500, 5xx}